src/Repository/Schedule/ScheduleRepository.php

```diff
@@ -0,0 +1,71 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Repository\Schedule;
+
+use App\Dto\ScheduleRequest;
+use App\Entity\Schedule;
+use Doctrine\ORM\EntityManagerInterface;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+final class ScheduleRepository implements ScheduleRepositoryInterface
+{
+    private $entityManager;
+    private $repository;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+        $this->repository = $entityManager->getRepository(Schedule::class);
+    }
+
+    public function find(string $id): ?Schedule
+    {
+        return $this->repository->find($id);
+    }
+
+    public function findAll()
+    {
+        return $this->repository->findAll();
+    }
+
+    /**
+     * @param ScheduleRequest $scheduleRequest
+     * @return Schedule
+     * @throws \Exception
+     */
+    public function createFrom(ScheduleRequest $scheduleRequest): Schedule
+    {
+        $schedule = new Schedule();
+        $schedule->setId($this->nextIdentity()->toString());
+        $schedule->setDay(
+            new \DateTime($scheduleRequest->day)
```

Comment:
Use a datetime immutable rather than a datetime, and either a DateTimeImmutable or a DateTimeInterface on the other side of this call

Reply:
Ok, done. #26